### PR TITLE
Replaced the translation language radio buttons in the options dialogue with a combo box

### LINKED ISSUE #6358: Language option as dropdown instead of radio buttons
We'll soon have more languages so I think the language choice in the Options dialog should be a dropdown combo box instead of radio buttons.

A non urgent simple task for someone to take up.

### PR DESCRIPTION
**Pull request by Ivanluv**
fixes #6358
@africanmathsinitiative/developers this is ready for review
@dannyparsons should I just Include the spanish and kiswahili options and have them enabled even though they are not working?

### CODE REVIEW

**Comment by lloyddewit** [comment]
@HawardKetoyoMsatsi As discussed, please could you do the first review, thanks

**Comment by HawardKetoyoMsatsi** [comment]
@lloyddewit Sure, I am happy to review this.

File: instat/dlgOptions.vb [hunk 1]
```diff
@@ -65,9 +65,6 @@ Public Class dlgOptions
         rtbOutputPreview.Text = strPreviewText
         SetView()
 
-        'temp disabled as not functioning yet
-        rdoSpanish.Enabled = False
-        rdoKiswahili.Enabled = False
```
[the remark this answers is not in the document]
Comment (Ivanluv) [reply instat/dlgOptions.vb:68]: thanks for noticing these.I have fixed it

File: instat/dlgOptions.vb [hunk 2]
```diff
@@ -360,7 +357,7 @@ Public Class dlgOptions
...
+    Private Sub AllControls_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrNudWaitSeconds.ControlValueChanged, ucrNudPreviewRows.ControlValueChanged, ucrNudMaxRows.ControlValueChanged, ucrNudMaxCols.ControlValueChanged, ucrNudDigits.ControlValueChanged, ucrNudAutoSaveMinutes.ControlValueChanged, ucrInputUserName.ControlValueChanged, ucrInputPort.ControlValueChanged, ucrInputHost.ControlValueChanged, ucrInputDatabaseName.ControlValueChanged, ucrInputComment.ControlContentsChanged, ucrChkViewProcurementMenu.ControlValueChanged, ucrChkViewOptionsByContextMenu.ControlValueChanged, ucrChkViewClimaticMenu.ControlValueChanged, ucrChkShowWaitDialog.ControlValueChanged, ucrChkShowSignifStars.ControlValueChanged, ucrChkShowRCommandsinOutputWindow.ControlValueChanged, ucrChkShowDataonGrid.ControlValueChanged, ucrChkIncludeDefaultParams.ControlValueChanged, ucrChkIncludeCommentsbyDefault.ControlValueChanged, ucrChkAutoSave.ControlValueChanged, ucrChkViewStructuredMenu.ControlValueChanged
 
     End Sub
```
[the remark this answers is not in the document]
Comment (Ivanluv) [reply instat/dlgOptions.vb:361]: I dont think it should be empty

**Review by rdstern — CHANGES_REQUESTED** [review]
This looks nice and is much better than the old radio buttons.

I also am quite happy that there are additional languages that show our intent.  And we can see they don't work yet, because when you return to the dialogue the pull-down has gone back to English.  

There is one small problem.  If I move the French then that's fine it is in French.  If I move to Swahili from French it stays in French, but the pull-down says English.  I assume it should remain in French.

**Review by rdstern — DISMISSED** [review]

**Comment by lloyddewit** [comment]
@rdstern @Ivanluv As mentioned above, the dropdown shows languages that are not yet implemented, and shows them as enabled. This is inconsistet with our policy on menus. We only enable a menu option **after** the corresponding dialog has been implemented. 

Having enabled, non-existent languages is confusing to the user. It also means that we have to add special code that jumps back to English or French if one of the not-implemented languages is slected. I tested this and does not yet work correctly.

It would be much simpler just to list English and French for now. We should add other languages to the list only when the language is included. This can be done as part of the PR for the new language.

If @rdstern  has no objections, then please could @Ivanluv remove the not-implemented languages. I can then approve and merge.
Thanks

**Comment by lloyddewit** [comment]
@Ivanluv I discussed with @rdstern . 
If you change the list to just contain English, French and Portugese then I can approve and merge.
Please treat Portugese exactly the same as English and French (i.e. assume that it is implemented in the database). This is because @dannyparsons will create a Portugese machine translation before the next release.
Thanks

File: instat/dlgOptions.vb [hunk 3]
```diff
@@ -120,19 +114,13 @@ Public Class dlgOptions
...
             Case "fr-FR"
-                rdoFrench.Checked = True
-            Case "sw-KE"
-                rdoKiswahili.Checked = True
-                ' temp disabled as not functioning
-                'Case "es-ES"
-                '    rdoSpanish.Checked = True
-            Case Else
-                rdoEnglish.Checked = True
+                ucrInputLanguage.SetText("French")
```
Comment (lloyddewit) [inline instat/dlgOptions.vb:121]: Please could you add a `Case`:
```
            Case "pt-PT"
                ucrInputLanguage.SetText("Portugese")
```

File: instat/dlgOptions.vb [hunk 4]
```diff
@@ -252,19 +240,16 @@ Public Class dlgOptions
...
-        End If
-        ApplyEnabled(True)
+    End Sub
+
+    Private Sub ucrInputLanguage_ControlValueChanged() Handles ucrInputLanguage.ControlValueChanged
+        Select Case ucrInputLanguage.GetText
+            Case "English"
+                strCurrLanguageCulture = "en-GB"
+            Case "French"
+                strCurrLanguageCulture = "fr-FR"
```
Comment (lloyddewit) [inline instat/dlgOptions.vb:250]: Please could you add a `Case`:
```
            Case "Portugese"
                strCurrLanguageCulture = "pt-PT"
```

File: instat/dlgOptions.vb (hunk 2 above)
[the remark this answers is not in the document]
Comment (lloyddewit) [reply instat/dlgOptions.vb:350]: I don't think this `Sub` has any purpose, please could you delete it?

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by lloyddewit** [comment]
@Ivanluv Please could you also resolve the merge conflicts? thanks

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Please could you retest/approve? thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
@Ivanluv just a tiny thing - please add the u in Portuguese.

File: instat/dlgOptions.vb [hunk 5]
```diff
@@ -1,414 +1,399 @@
...
+        ucrChkIncludeDefaultParams.SetText("Include Default Parameter Values in R Commands")
+        ucrChkAutoSave.SetText("Auto save a backup of data")
+        ucrChkShowWaitDialog.SetText("Show waiting dialog when command takes longer than")
+        ucrChkAutoSave.AddToLinkedControls(ucrNudAutoSaveMinutes, {True}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+        ucrNudAutoSaveMinutes.SetLinkedDisplayControl(lblMinutes)
+        ucrPnlGraphDisplay.AddRadioButton(rdoDisplayinOutputWindow)
+        ucrPnlGraphDisplay.AddRadioButton(rdoDisplayinRViewer)
+        ucrPnlGraphDisplay.AddRadioButton(rdoDisplayinSeparateWindows)
+        ucrInputLanguage.SetLinkedDisplayControl(lblLanguage)
+        ucrInputLanguage.SetItems({"English", "French", "Portugese"})
```
Comment (lloyddewit) [inline instat/dlgOptions.vb:86]:
```suggestion
        ucrInputLanguage.SetItems({"English", "French", "Portuguese"})
```

File: instat/dlgOptions.vb [hunk 6]
```diff
@@ -1,414 +1,399 @@
...
+        ucrInputHost.SetName(frmMain.clsInstatOptions.strClimsoftHost)
+        ucrInputPort.SetName(frmMain.clsInstatOptions.strClimsoftPort)
+        ucrInputUserName.SetName(frmMain.clsInstatOptions.strClimsoftUsername)
+        Select Case frmMain.clsInstatOptions.strLanguageCultureCode
+            Case "en-GB"
+                ucrInputLanguage.SetText("English")
+            Case "fr-FR"
+                ucrInputLanguage.SetText("French")
+            Case "pt-PT"
+                ucrInputLanguage.SetText("Portugese")
```
Comment (lloyddewit) [inline instat/dlgOptions.vb:123]:
```suggestion
                ucrInputLanguage.SetText("Portuguese")
```

File: instat/dlgOptions.vb [hunk 7]
```diff
@@ -1,414 +1,399 @@
...
+        Cursor = Cursors.Default
+    End Sub
+
+    Private Sub ucrInputLanguage_ControlValueChanged() Handles ucrInputLanguage.ControlValueChanged
+        Select Case ucrInputLanguage.GetText
+            Case "English"
+                strCurrLanguageCulture = "en-GB"
+            Case "French"
+                strCurrLanguageCulture = "fr-FR"
+            Case "Portugese"
```
Comment (lloyddewit) [inline instat/dlgOptions.vb:254]:
```suggestion
            Case "Portuguese"
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by lloyddewit** [comment]
> 
> 
> @Ivanluv just a tiny thing - please add the u in Portuguese.

@rdstern 
That was my fault - sorry!
I made the corrections, please could you approve? thanks

**Review by lloyddewit — APPROVED** [review]

**Review by rdstern — APPROVED** [review]

**Comment by dannyparsons** [comment]
@Ivanluv The combo box should be non-editable. Can you please update this?

**Comment by Ivanluv** [comment]
@dannyparsons  I have fixed this via PR #6498

**Pull request merged**